homeassistant/components/homeassistant/__init__.py

```diff
@@ -254,12 +255,28 @@ async def async_set_location(call: ha.ServiceCall) -> None:
             latitude=call.data[ATTR_LATITUDE], longitude=call.data[ATTR_LONGITUDE]
         )
 
+        service_data = {
+            "latitude": call.data[ATTR_LATITUDE],
+            "longitude": call.data[ATTR_LONGITUDE],
+        }
+
+        if call.data.get(ATTR_ELEVATION):
+            service_data["elevation"] = call.data[ATTR_ELEVATION]
+
+        await hass.config.async_update(**service_data)
```

Comment:
Don't think we need to update config data twice.
Check first and then call `async_update` once.